sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -1067,10 +1155,10 @@ def GetProvisionInfo(self, request, context=None):
         info=self._info)
 
 
-class GrpcWorkerHandler(WorkerHandler):
-  """An grpc based controller for fn API control, state and data planes."""
+class GrpcServer(object):
 
   _DEFAULT_SHUTDOWN_TIMEOUT_SECS = 5
+  _cached_grpc_server = None
```

Comment:
This is the one singleton global instance we should have. The fact that it's a singleton is just an optimization, hidden in its own `get_instance` method (which could return new instances each time, or some other strategy, if desired). All other state should be referenced off of its instance state (transitively).

Reply:
I didn't quite get it, can you please explain a little more?